cpp/src/arrow/vendored/datetime/tz.cpp

```diff
@@ -202,7 +202,12 @@ get_known_folder(const GUID& folderid)
     if (SUCCEEDED(hr))
     {
         co_task_mem_ptr folder_ptr(pfolder);
-        folder = std::string(folder_ptr.get(), folder_ptr.get() + wcslen(folder_ptr.get()));
+        folder.clear();
+        folder.reserve(wcslen(folder_ptr.get()));
+        for (std::size_t idx = 0; idx != wcslen(folder_ptr.get()); ++idx)
```

Comment:
This wcslen should be extracted as a separate variable to avoid doing a wcslen on every character.